### PR TITLE
Feature: Integrate with keycloak client credentials flow

### PR DESCRIPTION
**Pull request by topikachu**
The ui-idm-logic can use client credentials flow to retrieve access token.

#### Check List:
* Unit tests: NA
* Documentation: NA

### CODE REVIEW

File: modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java [hunk 1]
```diff
@@ -55,7 +59,11 @@
     @Override
     public void afterPropertiesSet() throws Exception {
         Assert.hasText(server, "server must be set");
-        Assert.hasText(authenticationPassword, "authenticationPassword must be set");
+        if (isClientCredentials()){
```
Comment (dbmalkovsky) [inline modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java:62]: small nitpick: add space before the brace `{`

File: modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java [hunk 2]
```diff
@@ -204,12 +232,29 @@ public AccessToken fetchAccessToken() {
             }
         }
 
+        private MultiValueMap<String, String> createAccessTokenRequestParameters() {
+            MultiValueMap<String, String> tokenRequestBody = new LinkedMultiValueMap<>();
+            tokenRequestBody.add("grant_type", getGrantType());
+            tokenRequestBody.add("client_id", getClientId());
+            if (isClientCredentials()){
```
Comment (dbmalkovsky) [inline modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java:239]: small nitpick: add space before the brace `{`

File: modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java (hunk 1 above)
Comment (topikachu) [reply modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java:62]: Shall I fix in a new commit or ammed the old one?

Comment (dbmalkovsky) [reply modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java:62]: you can just fix it in a new commit to this PR

Comment (topikachu) [reply modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java]: Done

File: modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java (hunk 2 above)
Comment (topikachu) [reply modules/flowable-ui/flowable-ui-idm-logic/src/main/java/org/flowable/ui/idm/service/keycloak/KeycloakConfiguration.java]: Done

**Pull request closed without merging**